# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmTypes.cpp
# Commit: 8ce9bbcd26e3
# Full Hash: 8ce9bbcd26e343a8d845a817cea33fd875805541
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2020-08-19 10:01:16
# Regressor Bug: 1561521
# File Overlap Count: 2
# Description:
#   Bug 1561521 - Generalize table/element code for future reference-types. r=lth
#   
#   wasm::Table uses wasm::TableKind to determine the element type of the table,
#   which makes it difficult to extend tables to support more reference types. This
#   commit drops TableKind in favor of directly storing the element RefType, along
# ==============================================================================

diff -r 7805ddfe2bfa -r 8ce9bbcd26e3 js/src/wasm/WasmTypes.cpp
--- a/js/src/wasm/WasmTypes.cpp	Tue Aug 18 16:58:32 2020 +0000
+++ b/js/src/wasm/WasmTypes.cpp	Tue Aug 18 16:58:48 2020 +0000
@@ -495,14 +495,14 @@
 }
 
 size_t ElemSegment::serializedSize() const {
-  return sizeof(kind) + sizeof(tableIndex) + sizeof(elementType) +
+  return sizeof(kind) + sizeof(tableIndex) + sizeof(elemType) +
          sizeof(offsetIfActive) + SerializedPodVectorSize(elemFuncIndices);
 }
 
 uint8_t* ElemSegment::serialize(uint8_t* cursor) const {
   cursor = WriteBytes(cursor, &kind, sizeof(kind));
   cursor = WriteBytes(cursor, &tableIndex, sizeof(tableIndex));
-  cursor = WriteBytes(cursor, &elementType, sizeof(elementType));
+  cursor = WriteBytes(cursor, &elemType, sizeof(elemType));
   cursor = WriteBytes(cursor, &offsetIfActive, sizeof(offsetIfActive));
   cursor = SerializePodVector(cursor, elemFuncIndices);
   return cursor;
@@ -511,7 +511,7 @@
 const uint8_t* ElemSegment::deserialize(const uint8_t* cursor) {
   (cursor = ReadBytes(cursor, &kind, sizeof(kind))) &&
       (cursor = ReadBytes(cursor, &tableIndex, sizeof(tableIndex))) &&
-      (cursor = ReadBytes(cursor, &elementType, sizeof(elementType))) &&
+      (cursor = ReadBytes(cursor, &elemType, sizeof(elemType))) &&
       (cursor = ReadBytes(cursor, &offsetIfActive, sizeof(offsetIfActive))) &&
       (cursor = DeserializePodVector(cursor, &elemFuncIndices));
   return cursor;